lib/devise_invalidatable/hooks/invalidatable.rb

```diff
@@ -6,7 +6,7 @@
 # easier to just use our own id.
 Warden::Manager.after_set_user except: :fetch do |user, warden, _|
   UserSession.deactivate(warden.raw_session['auth_id'])
-  warden.raw_session['auth_id'] = user.activate_session
+  warden.raw_session['auth_id'] = user.activate_session(warden)
```

Comment:
I feel like passing `warden` here is not a good thing: it makes `activate_session` depend on some internal state of `warden`.
Much better is to pass a hash of options, like this: `user.activate_session(ip: ..., user_agent: ...)`